From d06c41aef2272ac95387c69121fa3a47513b248b Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 26 Feb 2024 18:10:21 +0100
Subject: [PATCH 0676/1695] plat-stm32mp2: conf: disable CRYP when SAES is
 enabled.

override value of CFG_STM32_CRYP to n when CFG_STM32_SAES=y.
OP-TEE crypto framework can only register one AES accelerator.
If CRYP and SAES are probed, OP-TEE will panic.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 281f079a35b2aadf393c0a31af7caaf1ff49885b plat-stm32mp2: conf: disable CRYP when SAES is enabled.]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ie27a5575c0a2040dce156e6252f1239972bd504b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/363035
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 8944aa5d2..3cd009255 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -173,6 +173,12 @@ ifeq ($(CFG_STM32_HSE_MONITORING),y)
 $(call force,CFG_STM32_LPTIMER,y)
 endif
 
+#SAES and CRYP cannot be register at the same time in the crypto framework
+#If CRYP and SAES are enable, disable CRYP for safety purpose
+ifeq ($(call cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES), y)
+override CFG_STM32_CRYP := n
+endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
+
 # If any crypto driver is enabled, enable the crypto-framework layer
 ifeq ($(call cfg-one-enabled, CFG_STM32_SAES),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
-- 
2.47.2

